node/src/hopper/routing_service.rs

```diff
@@ -1869,6 +1860,113 @@ mod tests {
         );
     }
 
+    fn route_data_to_peripheral_component_uses_proper_key_on_payload_for_component<F>(
+        payload_factory: F,
+        target_component: Component,
+    ) where
+        F: FnOnce(&CryptDEPair) -> CryptData,
+    {
+        let peer_actors = peer_actors_builder().build();
+        let subject = RoutingService::new(
+            make_cryptde_pair(),
+            RoutingServiceSubs {
+                proxy_client_subs_opt: peer_actors.proxy_client_opt,
+                proxy_server_subs: peer_actors.proxy_server,
+                neighborhood_subs: peer_actors.neighborhood,
+                hopper_subs: peer_actors.hopper,
+                to_dispatcher: peer_actors.dispatcher.from_dispatcher_client,
+                to_accountant_routing: peer_actors.accountant.report_routing_service_provided,
+            },
+            100,
+            200,
+            true,
+        );
+        let route = Route::single_hop(&PublicKey::new(b"1234"), subject.cryptdes.main).unwrap();
+        let payload = payload_factory(&subject.cryptdes);
+        let live_package = LiveCoresPackage::new(route, payload);
+
+        subject.route_data_to_peripheral_component(
+            target_component,
+            SocketAddr::from_str("1.2.3.4:1234").unwrap(),
+            live_package,
+            true,
+        );
+
+        // CryptDENull panics when you try decrypting with the wrong key; no panic means test passes
+    }
+
+    #[test]
+    fn route_data_to_peripheral_component_uses_alias_key_on_payload_for_proxy_client() {
+        let payload_factory = |cryptdes: &CryptDEPair| {
+            encodex(
+                cryptdes.alias,
+                cryptdes.main.public_key(),
```

Comment:
Why are those two supplied cryptDEs each of a different kind? I wouldn't have expected to see the alias cryptDE used anywhere here, that is on the route over...